<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1024</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1024</width>
    <height>720</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTableWidget" name="processTable">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>30</y>
      <width>871</width>
      <height>261</height>
     </rect>
    </property>
    <property name="horizontalScrollBarPolicy">
     <enum>Qt::ScrollBarPolicy::ScrollBarAlwaysOff</enum>
    </property>
    <property name="autoScroll">
     <bool>true</bool>
    </property>
    <column>
     <property name="text">
      <string>Process ID</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Arrival Time</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Burst Time</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Priority</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Completion Time</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Turnaround Time</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Waiting Time</string>
     </property>
    </column>
   </widget>
   <widget class="QPushButton" name="calculateButton">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>310</y>
      <width>83</width>
      <height>29</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculate</string>
    </property>
   </widget>
   <widget class="QSplitter" name="splitter">
    <property name="geometry">
     <rect>
      <x>920</x>
      <y>120</y>
      <width>71</width>
      <height>83</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Vertical</enum>
    </property>
    <widget class="QPushButton" name="addButton">
     <property name="autoFillBackground">
      <bool>false</bool>
     </property>
     <property name="text">
      <string>+</string>
     </property>
    </widget>
    <widget class="QPushButton" name="deleteButton">
     <property name="text">
      <string>-</string>
     </property>
    </widget>
    <widget class="QPushButton" name="resetButton">
     <property name="text">
      <string>üóëÔ∏è</string>
     </property>
    </widget>
   </widget>
   <widget class="QSplitter" name="splitter_2">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>310</y>
      <width>224</width>
      <height>24</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Orientation::Horizontal</enum>
    </property>
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Enter Time Quantum: </string>
     </property>
    </widget>
    <widget class="QLineEdit" name="timeQuantumInput"/>
   </widget>
   <widget class="QGraphicsView" name="ganttChartView">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>360</y>
      <width>961</width>
      <height>231</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="algorithmComboBox">
    <property name="geometry">
     <rect>
      <x>721</x>
      <y>310</y>
      <width>161</width>
      <height>28</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Round Robin</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Preemptive SJF</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Non Preemptive SJF</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Preemptive Priority</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Non Preemptive Priority</string>
     </property>
    </item>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>600</y>
      <width>271</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Total Turnaround Time:      </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="totalTurnaroundTimeOutput">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>Average Turnaround Time: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="averageTurnaroundTimeOutput">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget_2">
    <property name="geometry">
     <rect>
      <x>70</x>
      <y>600</y>
      <width>251</width>
      <height>71</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Total Waiting Time:      </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="totalWaitingTimeOutput">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Average Waiting Time: </string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="averageWaitingTimeOutput">
         <property name="enabled">
          <bool>false</bool>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
